### ISSUE TITLE
Groups feature

### ISSUE THREAD

**Issue by n3storm** [nitely/Spirit#88]
Hi, Spirit looks promising, thanks nitely!

Is there anybody interested in Groups? 
Groups of users can be public or private.
Users can only participate in groups where they are members.

Ultimate goal is to emulate http://groupserver.org/ including participating via email.

**Comment by nitely** [comment]
Hi :)

Those "groups" seem like categories to me. The only difference I can spot is users can subscribe or join to groups/categories and also create them. It should be possible to extend Spirit that way. But currently it's out of the *core* scope (of being just a regular forum). Maybe a extension/plugin can be created for such purpose.

**Comment by n3storm** [comment]
I agree categories could be used for this purpose though name can be confusing for users.

There could be a way to add users and moderators to a category like in private chats.
I would like to provide a Middleware to make Categories private so only suscribed users can see/access them.
Another middleware would make whole site private.

Should I name the plugin in some specially way? spirit_groupsmode is ok for you?

Thanks.

**Comment by nitely** [comment]
> Another middleware would make whole site private.

That [middleware is implemented](https://github.com/nitely/Spirit/blob/master/spirit/core/middleware.py#L17)

> Should I name the plugin in some specially way? spirit_groupsmode is ok for you?

Yep, that should be ok.

**Comment by alesdotio** [comment]
I implemented something similar on my fork https://github.com/alesdotio/Spirit/tree/master/spirit/group
It's just an admin for django's Group model, which you can assign to categories https://github.com/alesdotio/Spirit/blob/master/spirit/category/models.py#L31 which will in turn make them hidden/readonly/etc.

I asked about this in #67 What do you think @nitely?

**Comment by nitely** [comment]
> I asked about this in #67 What do you think @nitely?

I don't have an opinion yet, I'd like to experiment different approaches (ie: I'd like to try the built-in permissions), that's why I didn't say anything about #67.

**Comment by n3storm** [comment]
@alesdotio your fork does exactly what I was about to code! thanks for the headsup
@nitley thanks for noticing the middleware is already created. For the documentation, set ST_PRIVATE_FORUM = True in your settings/base.py and Spirit will always ask your users to login before seeing anything else.

**Comment by thomasyip** [comment]
I made a proposal on object level support, which would ultimately solves the group problem.

Want to see if anyone interested: https://github.com/nitely/Spirit/issues/99 

Perhaps we can pool in some effort etc.
